crates/winmd/src/types/delegate.rs

```diff
@@ -152,7 +153,7 @@ impl Delegate {
                     interface: *mut ::winrt::RawPtr,
                 ) -> ::winrt::ErrorCode {
                     unsafe {
-                        let this = this as *const Self as *mut Self;
+                        let this: *mut Self = this.map(|s| s.cast::<Self>()).unwrap().as_ptr();
```

Comment:
Do we know how much more expensive this pattern is at run time?

Reply:
It's the same except it adds a null check and a possible branch to a panic handler (due to the unwrap). Of course, if the ptr is null in the original code, who knows what will actually happen.

https://godbolt.org/z/6zNWvN